domain/relation/state/types.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state
+
+import (
+	"github.com/juju/juju/internal/charm"
+	"github.com/juju/juju/internal/relation"
+)
+
+type relationUUID struct {
+	UUID string `db:"uuid"`
+}
+
+type relationIDAndUUID struct {
+	// UUID is the UUID of the relation.
+	UUID string `db:"uuid"`
+	// ID is the numeric ID of the relation
+	ID int `db:"relation_id"`
+}
+
+// endpoint is used to fetch endpoints from the database.
```

Comment:
nit: s/fetch endpoints/fetch an endpoint/

this is a struct not a slice of structs.